src/App.tsx

```diff
@@ -2,13 +2,24 @@ import React from 'react';
 
 import './App.scss';
 
-// import posts from './api/posts';
-// import comments from './api/comments';
+import posts from './api/posts';
+import users from './api/users';
+import comments from './api/comments';
+import { PostList } from './components/PostList/PostList';
+import { Post } from './types/types';
 // import users from './api/users';
 
+const preparedPosts: Post[] = posts.map(post => ({
+  ...post,
+  user: users.find((user) => user.id === post.userId) || null,
+  comments: comments.filter(comment => comment.postId === post.id) || [],
```

Comment:
filter always returns an array. If don't find matches, returns an empty array

```suggestion
  comments: comments.filter(({ postId })=> postId === post.id),
```